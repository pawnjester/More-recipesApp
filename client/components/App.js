import React from 'react';
import style from '../styles/index.scss';

export default () => {
  return (
    <h1>Hello World</h1>
  );
}