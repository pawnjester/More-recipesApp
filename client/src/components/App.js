import React from 'react';
import '../styles/index.scss';
import NavigationBar from './NavigationBar';
import Main from './Main';
import FlashMessagesList from './flash/FlashMessagesList';

class App extends React.Component {
  render() {
    return (
      <div className= 'container'>
        <NavigationBar />
        <FlashMessagesList />
        <Main />
      </div>
    );
  }
}

export default App;