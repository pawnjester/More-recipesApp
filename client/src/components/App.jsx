import React from 'react';
import Routes from './Routes';

const App = () => (
  <div className="wrap">
    <Routes />
  </div>
);


export default App;
