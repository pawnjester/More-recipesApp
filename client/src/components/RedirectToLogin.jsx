import React from 'react';
import { Link } from 'react-router-dom';
import NavigationBarComponent from './NavigationBar';


export const RedirectToLogin = () => (
  <div>
    <NavigationBarComponent />
    <div className="container">
      <div className="row">
        <div className="card mt-5" style={{ marginLeft: 300 }}>
          <div card-body="card-body" >
            <p className="mt-5 p-5">
              Your Password has been changed,
              <br />
              <br />
              Please click this
              <Link to="/signin">link
              </Link> to login
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
);


export default RedirectToLogin;
