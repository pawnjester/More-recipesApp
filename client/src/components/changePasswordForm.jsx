import React, { Component } from 'react';
import { connect } from 'react-redux';
import toastr from 'toastr';
import NavigationBar from './NavigationBar';
import changePassword from '../actions/changePassword';
import '../styles/passwordchange.scss';

/**
 *
 *
 * @class ChangePasswordForm
 *
 * @extends {Component}
 */

class ChangePasswordForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      oldPassword: '',
      password: ''
    };
  };

  onNameChange = (event) => {
    this.setState({ [event.target.name]: event.target.value });
  };

  onSubmit = (event) => {
    event.preventDefault();
    this.props.changePassword(this.state).then((res) => {
      toastr.success('Password successfully changed');
      this.setState({
        oldPassword: '',
        password: ''
      });
    })
      .catch(error => toastr.error(error.response.data.error));
  }

  render() {
    return (
      <div >
        <NavigationBar />
        <div>
          <form className="mx-auto app-login-form card-login">
            <h4 className="black text-center mt-5">Change Your Password</h4>
            <section className="pt-4 input-group">
              <span className="input-group-addon">
                <i className="fa fa-lock black" aria-hidden="true"></i>
              </span>
              <input
                name="oldPassword"
                className="form-control"
                type="password"
                value={this.state.oldPassword}
                onChange={this.onNameChange}
                placeholder="Enter your old password"
              />
            </section>
            <section className="pt-4 pb-3 input-group">
              <span className="input-group-addon">
                <i className="fa fa-lock black" aria-hidden="true"></i>
              </span>
              <input
                name="password"
                className="form-control"
                type="password"
                value={this.state.password}
                onChange={this.onNameChange}
                placeholder="Enter a new Password"
              />
            </section>
            <div className="pt-2">
              <button
                className="btn btn-block text-capitalize password-color"
                onClick={this.onSubmit}
              >
                change Password
              </button>
            </div>
          </form>
        </div>
      </div>
    );
  }
}

export default connect(null, { changePassword })(ChangePasswordForm);
