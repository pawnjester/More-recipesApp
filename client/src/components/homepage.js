import React from 'react';
import '../styles/home.scss';
import Signupform from './signup/signupform';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { userSignupRequest } from '../actions/signupActions';
import { Button } from 'react-bootstrap';



class Home extends React.Component {
  render() {
    const {userSignupRequest} = this.props;

    return (
    <header id ="home-section">
      <div className= "dark-overlay">
      <div className= "home-inner">
        <div className ="container">
          <div className= "row">
            <div className="col-lg-8 d-none d-lg-block">
              <h1 className= "display-4" style = {{color: 'fff'}}>Let the  world know about your <strong>Recipe</strong> in three easy steps:</h1>
              <div className="d-flex flex-row">
                <div className ="p-4 align-self-start" >
                  <i className="fa fa-check"></i>
                  
                </div>
                <div className= "p-4 align-self-end" style = {{color: 'fff'}}>
                  Create an account
                </div>
                
              </div>

              <div className="d-flex flex-row">
                <div className ="p-4 align-self-start" >
                  <i className="fa fa-check"></i>
                  
                </div>
                <div className= "p-4 align-self-end" style = {{color: 'fff'}}>
                  Add A Recipe 
                </div>
                
              </div>

              <div className="d-flex flex-row">
                <div className ="p-4 align-self-start" >
                  <i className="fa fa-check"></i>
                  
                </div>
                <div className= "p-4 align-self-end" style = {{color: 'fff'}}>
                  Relax, and let the world view your wonderful recipe
                </div>
                
              </div>
            </div>
            <div className="col-lg-4">
              <div className= "card bg-danger text-center card-form">
                <div className="card-body">
                  <h3>Sign Up Today</h3>
                  <p>Please fill this form to register</p>
                    <Signupform userSignupRequest = {userSignupRequest} />
                </div>
              </div>
            </div>
          </div>
          
        </div>
        
      </div>
      </div>
      
    </header >
    )
  }
}

Home.propTypes = {
  userSignupRequest: PropTypes.func.isRequired,
}

export default connect(null, { userSignupRequest})(Home)