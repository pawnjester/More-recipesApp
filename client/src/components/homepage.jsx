import React from 'react';
import Signupform from './signup/signupform';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { userSignupRequest } from '../actions/signupActions';
import { Button } from 'react-bootstrap';
import NavigationBar from './NavigationBar';
import '../styles/home.scss';
/*eslint-disable */


class Home extends React.Component {
  render() {
    const { userSignupRequest } = this.props;

    return (
      <div>
        <NavigationBar />
        <header id="home-section">
          <div className="dark-overlay">
            <div className="home-inner">
              <div className="container">
                <div className="row">
                  <div className="col-lg-8 d-none d-lg-block">
                <h1 className="display-4" style={{ color: 'fff' }}>Let the  world know about your <strong>Recipe</strong> in three easy steps:</h1>
                <div className="d-flex flex-row">
                  <div className="p-4 align-self-start" >
                  <i className="fa fa-check back" />

                </div>
                  <div className="p-4 align-self-end ">
                  Create an account
                </div>

                </div>

                <div className="d-flex flex-row">
                  <div className="p-4 align-self-start" >
                  <i className="fa fa-check back" />

                </div>
                  <div className="p-4 align-self-end ">
                  Add A Recipe
                </div>

                </div>

                <div className="d-flex flex-row">
                  <div className="p-4 align-self-start" >
                  <i className="fa fa-check back" />

                </div>
                  <div className="p-4 align-self-end">
                  Relax, and let the world view your wonderful recipe
                </div>

                </div>
              </div>
                  <div className="col-lg-4">
                <div className="card bg-danger text-center card-form">
                  <div className="card-body">
                  <h3>Sign Up Today</h3>
                  <p>Please fill this form to register</p>
                  <Signupform userSignupRequest={userSignupRequest} />
                </div>
                </div>
              </div>
                </div>

              </div>

            </div>
          </div>

        </header >
      </div>
    );
  }
}

Home.propTypes = {
  userSignupRequest: PropTypes.func.isRequired,
};

export default connect(null, { userSignupRequest })(Home);
