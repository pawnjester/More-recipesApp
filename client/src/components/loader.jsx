import React from 'react';
import { RingLoader } from 'react-spinners';


const Loader = props => (
  <div className="sweet-loading loader-center">
    <RingLoader
      color="#e50000"
      loading={props.loading}
    />
  </div>
);

export default Loader;
