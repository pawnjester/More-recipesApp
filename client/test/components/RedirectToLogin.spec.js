import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import expect from 'expect';
import { RedirectToLogin } from '../../src/components/RedirectToLogin';


configure({ adapter: new Adapter() });

const setup = () => {
  const shallowWrapper = shallow(<RedirectToLogin />);
  return {
    shallowWrapper,
  };
};

describe('Redirect to Login Page', () => {
  it('should render correctly', () => {
    const { shallowWrapper } = setup();
    expect(shallowWrapper).toMatchSnapshot();
  });
});
