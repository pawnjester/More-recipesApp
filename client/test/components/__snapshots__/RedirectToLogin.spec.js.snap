// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redirect to Login Page should render correctly 1`] = `
<div>
  <Connect(NavigationBar) />
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="card mt-5"
        style={
          Object {
            "marginLeft": 300,
          }
        }
      >
        <div
          card-body="card-body"
        >
          <p
            className="mt-5 p-5"
          >
            Your Password has been changed,
            <br />
            <br />
            Please click this 
            <Link
              replace={false}
              to="/signin"
            >
              link
            </Link>
             to login
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
